create_full_label_visualizations.py

```diff
@@ -0,0 +1,308 @@
+    
+    for node in important_hop2:
+        name = str(G.nodes[node]['name']) if G.nodes[node]['name'] else node
+        # FULL names for important 2-hop neighbors too
+        labels[node] = name
+    
+    # Draw labels with adjusted formatting for full names
+    nx.draw_networkx_labels(G, pos, labels, font_size=7, font_weight='bold', 
+                           font_color='black', bbox=dict(boxstyle='round,pad=0.3', 
+                           facecolor='white', alpha=0.9, edgecolor='gray'))
+    
+    # Add title
+    center_name_full = str(G.nodes[center_id]['name']) if G.nodes[center_id]['name'] else center_id
+    plt.title(f'2-Hop Subgraph around {center_name_full} - FULL LABELS\n({len(G.nodes)} nodes, {len(G.edges)} edges)', 
+              fontsize=26, fontweight='bold', pad=60)
+    
+    # Add legend
+    legend_elements = []
+    for category, color in category_colors.items():
+        if any(category in str(data.get('category', '')) for _, data in G.nodes(data=True)):
+            legend_elements.append(plt.Line2D([0], [0], marker='o', color='w', 
+                                            markerfacecolor=color, markersize=15, 
+                                            label=category.replace('biolink:', '')))
+    
+    # Add hop legend
+    legend_elements.append(plt.Line2D([0], [0], marker='o', color='w', 
+                                    markerfacecolor='black', markersize=20, 
+                                    label='Center (Hop 0)'))
+    legend_elements.append(plt.Line2D([0], [0], marker='o', color='w', 
+                                    markerfacecolor='black', markersize=15, 
+                                    label='1-Hop Neighbors'))
+    legend_elements.append(plt.Line2D([0], [0], marker='o', color='w', 
+                                    markerfacecolor='black', markersize=10, 
+                                    label='2-Hop Neighbors'))
+    
+    plt.legend(handles=legend_elements, loc='upper left', bbox_to_anchor=(1.02, 1), 
+              fontsize=14, frameon=True, fancybox=True, shadow=True)
+    
+    # Remove axes
+    plt.axis('off')
+    
+    # Adjust layout and save
+    plt.tight_layout()
+    output_file = '/Users/marcin/Documents/VIMSS/ontology/KG-Hub/KG-Microbe/MicroGrowLink/evaluation_results/corynebacterium_glutamicum_2hop_subgraph_full_labels.png'
+    plt.savefig(output_file, dpi=300, bbox_inches='tight', facecolor='white')
+    plt.show()
+    
+    print(f"2-hop visualization with full labels saved to {output_file}")
+
+def main():
+    print("Creating 1-hop subgraph with full labels...")
+    create_1hop_full_labels()
+    
+    print("\nCreating 2-hop subgraph with full labels...")
+    create_2hop_full_labels()
+    
+    print("\nBoth visualizations with full labels have been created!")
+    print("Files saved to: /Users/marcin/Documents/VIMSS/ontology/KG-Hub/KG-Microbe/MicroGrowLink/evaluation_results/")
```

Comment:
are you going to check the output of processes like this into the repo?

Reply:
I'm updating to the local path here.. but the only time I would do that is if there is a specific reusable data product (eg list of unmapped chemicals).